pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -728,7 +728,7 @@ PJ_DEF(pj_status_t) pjsua_reconfigure_logging(const pjsua_logging_config *cfg)
     pjsua_logging_config_dup(pjsua_var.pool, &pjsua_var.log_cfg, cfg);
 
     /* Redirect log function to ours */
-    pj_log_set_log_func( &log_writer );
+    pj_log_set_log_func( (cfg && cfg->cb && !cfg->log_filename.slen) ? cfg->cb : &log_writer );
```

Comment:
- We missed `cfg != NULL` verification, better add that, e.g: `PJ_ASSERT_RETURN(cfg, PJ_EINVAL);`
- Re: skipping the `log_writer`, it may change the behavior of the lib (potential backward compatibility issue), could you specify the reason?

Reply:
Done for the first one, the second one reverted to current code.

Reply:
We need to pass a different log_writer call back function that write pj logs into our own application logs. how are we going to achieve that without this implementation.

Reply:
The function log_writer() will call you callback.

Reply:
this is an optimization we added. it directly call 'cfg->cb' if user has defined that; without calling it through log_writer. But it is OK to stick with current implementation.